crates/runtime/src/dataconnector/postgres.rs

```diff
@@ -0,0 +1,121 @@
+use async_trait::async_trait;
+use bb8_postgres::tokio_postgres::types::ToSql;
+use bb8_postgres::tokio_postgres::NoTls;
+use bb8_postgres::PostgresConnectionManager;
+use datafusion::datasource::TableProvider;
+use datafusion::sql::TableReference;
+use futures::TryStreamExt;
+use snafu::prelude::*;
+use sql_provider_datafusion::dbconnectionpool::postgrespool::PostgresConnectionPool;
+use sql_provider_datafusion::dbconnectionpool::DbConnectionPool;
+use sql_provider_datafusion::SqlTable;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::{collections::HashMap, future::Future};
+
+use arrow::array::RecordBatch;
+use spicepod::component::dataset::Dataset;
+
+use crate::secrets::Secret;
+
+use super::DataConnector;
+use super::Result;
+use super::UnableToGetTableProviderSnafu;
+
+pub struct Postgres {
+    pool: Arc<
+        dyn DbConnectionPool<
+                bb8::PooledConnection<'static, PostgresConnectionManager<NoTls>>,
+                &'static (dyn ToSql + Sync),
+            > + Send
+            + Sync,
+    >,
+}
+
+#[async_trait]
+impl DataConnector for Postgres {
+    fn new(
+        _secret: Option<Secret>,
```

Comment:
assuming this is going to be fixed once the store is integrated?

Reply:
Yes it will be used there. But it's part of the trait definition so we have to include it even if we aren't using it.